### PR TITLE
fix: gas calcuation function for scalar operations of TEE

### PR DESCRIPTION
**Pull request by lovenoble**
This is the fix for scalar operations of TEE.
In https://github.com/Inco-fhevm/fhevm-go/pull/23, I missed considering the scalar cases in calculating the gas.
In this PR, I fixed this one.

### CODE REVIEW

File: fhevm/tee_gas.go [hunk 1]
```diff
@@ -11,15 +12,30 @@ func teeOperationGas(op string, environment EVMEnvironment, input []byte, gasCos
 
 	logger := environment.GetLogger()
 
-	var lhs, rhs *verifiedCiphertext
-	lhs, rhs, err := get2VerifiedOperands(environment, input)
+	isScalar, err := isScalarOp(input)
 	if err != nil {
-		logger.Error(op, "RequiredGas() ciphertext inputs not verified", "err", err, "input", hex.EncodeToString(input))
+		logger.Error(fmt.Sprintf("%s can not detect if operator is meant to be scalar", op), "err", err, "input", hex.EncodeToString(input))
```
Comment (amaury1093) [inline fhevm/tee_gas.go]:
```suggestion
		logger.Error(fmt.Sprintf("%s can not detect if operator was meant to be scalar", op), "err", err, "input", hex.EncodeToString(input))
```

File: fhevm/tee_gas.go [hunk 2]
```diff
@@ -11,15 +12,30 @@ func teeOperationGas(op string, environment EVMEnvironment, input []byte, gasCos
...
+			return 0
+		}
+		if lhs.fheUintType() != rhs.fheUintType() {
+			logger.Error(op, "RequiredGas() operand type mismatch", "lhs", lhs.fheUintType(), "rhs", rhs.fheUintType())
+			return 0
+		}
+	} else {
+		lhs, _, err = getScalarOperands(environment, input)
+		if err != nil {
+			logger.Error("fheAdd scalar inputs not verified", "err", err, "input", hex.EncodeToString(input))
```
Comment (amaury1093) [inline fhevm/tee_gas.go]: fheAdd is not correct here

File: fhevm/tee_test.go [hunk 3]
```diff
@@ -21,12 +21,19 @@ func teeOperationHelper(t *testing.T, fheUintType tfhe.FheUintType, lhs, rhs, ex
...
+
+	var input []byte
+	if !isScalar {
+		rhsCt, err := importTeePlaintextToEVM(environment, depth, rhs, fheUintType)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+		input = toLibPrecompileInput(signature, false, lhsCt.GetHash(), rhsCt.GetHash())
+	} else {
+		valueBz, _ := marshalTfheType(rhs, fheUintType)
```
Comment (amaury1093) [inline fhevm/tee_test.go]: let's still Fatalf here, just in case

**Review by amaury1093 — APPROVED** [review]
Some nits, can merge after they are addressed

**Comment by amaury1093** [comment]
One request @lovenoble : could you:
- change the PR title to something more descriptive? I think this PR is a `fix:`, not a `feat:`
- Could you also add a description in the PR? At the minimum, link the other scalar PR, and give in 2-3 lines the changes you are making in this PR.

It will help making reviews easier.

File: fhevm/tee_gas.go (hunk 1 above)
Comment (lovenoble) [reply fhevm/tee_gas.go]: done

File: fhevm/tee_test.go (hunk 3 above)
Comment (lovenoble) [reply fhevm/tee_test.go]: done

File: fhevm/tee_gas.go (hunk 2 above)
Comment (lovenoble) [reply fhevm/tee_gas.go]: done

**Review by amaury1093 — APPROVED** [review]

**Pull request merged**